cfe_internal/enterprise/CFE_knowledge.cf

```diff
@@ -221,6 +221,22 @@ bundle agent cfe_internal_permissions
 # @brief Specific expectations for permissions and ownership of CFEngine with respect to the Enterprise Edition
 {
 
+  vars:
+
+    # nonstandard directories in statedir
+    "_statedir_standard_perm_exceptions"
+      slist => { "pg", "cf-execd.sockets" };
+
+    # Derive the users that should be able to access cf-execd sockets
+    "_cf_execd_socket_allow_users"
+      slist => { @(def.control_executor_runagent_socket_allow_users) };
+
+    # Derive the aces for users allowed to access cf-execd sockets
+    "_cf_execd_socket_dir_aces" -> { "ENT-6777" }
+      slist => maplist( "user:$(this):rx:allow", @(_cf_execd_socket_allow_users) );
+    "_cf_execd_socket_runagent_aces" -> { "ENT-6777" }
+      slist => maplist( "user:$(this):rw:allow", @(_cf_execd_socket_allow_users) );
```

Comment:
I've only just now found out that we need to set the `r-x` ACLs on the state directory too. *cfapache* and *cfpostgres* are special users for which this is not necessary because the state directory is group-owned by the *cfpostgres* group which includes the *cfapache* user. So these two users can access the state directory, but other users cannot.

Reply:
OK, I have added ACLs for sys.statedir https://github.com/cfengine/masterfiles/pull/1918/commits/2052313c121a65da677da49a31c9d0c8acf806cd